src/core/thread/mle_router.cpp

```diff
@@ -860,9 +866,10 @@ ThreadError MleRouter::HandleLinkAccept(const Message &aMessage, const Ip6::Mess
     case kDeviceStateChild:
         SuccessOrExit(error = Tlv::GetTlv(aMessage, Tlv::kLinkMargin, sizeof(linkMargin), linkMargin));
         VerifyOrExit(linkMargin.IsValid(), error = kThreadError_Parse);
-        mRouters[routerId].mLinkQualityOut = LinkMarginToQuality(linkMargin.GetLinkMargin());
-        mRouters[routerId].mLinkQualityIn = LinkMarginToQuality(reinterpret_cast<const ThreadMessageInfo *>
-                                                                (aMessageInfo.mLinkInfo)->mLinkMargin);
+        mRouters[routerId].mLinkQualityOut =
+            LinkQualityInfo::ConvertLinkMarginToLinkQuality(linkMargin.GetLinkMargin());
+        mRouters[routerId].mLinkQualityIn =
```

Comment:
Good point. I think we can replace/remove `mLinkQualityIn`. Let me check where/how it's used to make sure.